File: 1396-count-servers-that-communicate/count-servers-that-communicate.cpp
class Solution {
public:
    int countServers(vector<vector<int>>& grid) {
        vector<int> rowCount(grid.size(), 0);
        vector<int> colCount(grid[0].size(), 0);

        for(int i=0; i<grid.size(); i++){
            for(int j=0; j<grid[0].size(); j++){
                if (grid[i][j] == 1){
                    rowCount[i]++;
                    colCount[j]++;
                }
            }
        }
        int ans = 0;
        for(int i=0; i<grid.size(); i++){
            for(int j=0; j<grid[0].size(); j++){
                if (grid[i][j] == 1){
                    if (rowCount[i] > 1) ans += 1;
                    else if (colCount[j] > 1) ans += 1;
                }
            }
        }

        return ans;
    }
};

File: 1396-count-servers-that-communicate/count-servers-that-communicate.go
func countServers(grid [][]int) int {
	m, n := len(grid), len(grid[0])
	rows, cols := make([]int, m), make([]int, n)

	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			if grid[i][j] == 1 {
				rows[i]++
				cols[j]++
			}
		}
	}
	ans := 0
	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			if grid[i][j] == 1 {
				if rows[i] > 1 || cols[j] > 1 {
					ans++
				}
			}
		}
	}

	return ans
}